bokeh/plotting/helpers.py

```diff
@@ -170,7 +171,33 @@ def _update_legend(plot, glyph_renderer):
         legend = legends[0]
     else:
         raise RuntimeError("Plot %s configured with more than one legend renderer" % plot)
-    legend.legends.append(glyph_renderer)
+
+    label = glyph_renderer.glyph.label
+
+    if plot._compound_legend is True:
+        compound_legend = True
+    else:
+        compound_legend = False
+        # But test if we now need to switch
+        if label.get('value'):
+            for r in plot.renderers:
+                if isinstance(r, GlyphRenderer) and hasattr(r.glyph, 'label') and r.glyph.label == label:
```

Comment:
`label` is defined on `Glyph`, so every specific glyph class has this property. `hasattr()` is unnecessary.